### ISSUE TITLE
Roles Dont AutoStart After Reboot

### ISSUE THREAD

**Issue by preiner** [aws-quickstart/quickstart-confluent-kafka#47]
In cp-deploy.sh, there is a line that checks for the existence of init scripts, however, i have verified that they are not present

> if [ -x $CP_HOME/initscripts/cp-kafka-service ] ; then

Is there some reason they are missing from the tarball or am I missing something obvious?

**Comment by dbtucker** [comment]
The initscripts are not part of the Confluent distribution.    For environments where a fixed AMI was used (not ideal in the case of the quickstart because it restricts flexibility on operating systems and instance types), the initscripts were pre-installed.  

Additionally, the Quickstart does not currently support "stopping" the running instances as opposed to terminating the entire cluster.   While such an operation is possible with careful administration of the auto-scaling groups, it was beyond the scope of this initial effort.   Therefore, the need for service scripts in the deployed instances was relatively minor.

If you'd like to see examples of start-up scripts, there are several external repositories that are worth investigating: https://github.com/dbtucker/cp-service and https://github.com/jcustenborder/confluent-startup-scripts.

**Comment by preiner** [comment]
Terminating an entire cluster isnt viable for us because we require "exactly once" message processing. The issue of neither being able to reboot or terminate a instance due to hang, OS patching, ... seems to limit the solution to development / demo use cases. Could you elaborate on issues/edge cases we should pay attention to if we need to support terminating an instance?

**Comment by cyan-confluent** [comment]
@preiner You are correct. The scope of the current AWS template is limited to development and PoC clusters. The current templates do not cover all the long running operations tasks.

Even if init scripts are generated, there are still some common maintenance issues with operations, it's not always straightforward:
For example, if you frequently add / delete nodes, you will need to think about data balancing with those instances. In cases like that, you have to rely on the Auto-Data load balancer in the Confluent Enterprise packages.
